sql/logictest: TestLogic failed

[(sql/logictest).TestLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2648131&tab=buildLog) on [master@a941442bd0cb51083c0f3bb2c6bf2f1bd02285ec](https://github.com/cockroachdb/cockroach/commits/a941442bd0cb51083c0f3bb2c6bf2f1bd02285ec):

```
=== RUN   TestLogic
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic353535298
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestLogic
    logic.go:3316: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic353535298
--- FAIL: TestLogic (811.89s)
=== RUN   TestLogic/fakedist-disk/alter_type/add_value
--- done: testdata/logic_test/computed with config fakedist-disk: 168 tests, 0 failures
=== CONT  TestLogic/fakedist-disk/alter_type/add_value
    logic.go:1977: 
         
        testdata/logic_test/alter_type:375: 
        expected:
        pq: could not validate enum value removal for "d": count-value-usage: enum label "d" is not yet public
        got:
        pq: could not remove enum value "d" as it is being used by "v" in row: alphabets='d'
E210205 15:08:21.835703 8909753 sql/logictest/logic.go:3523  [-] 874  
E210205 15:08:21.835703 8909753 sql/logictest/logic.go:3523  [-] 874 +testdata/logic_test/alter_type:378: too many errors encountered, skipping the rest of the input
    logic.go:1809: 
         testdata/logic_test/alter_type:378: too many errors encountered, skipping the rest of the input
--- done: testdata/logic_test/alter_type with config fakedist-disk: 95 tests, 2 failures
            --- FAIL: TestLogic/fakedist-disk/alter_type/add_value (1.59s)
=== RUN   TestLogic/fakedist-disk/alter_type
=== PAUSE TestLogic/fakedist-disk/alter_type
=== CONT  TestLogic/fakedist-disk/alter_type
--- progress: testdata/logic_test/alter_primary_key: 49 statements/queries
        --- FAIL: TestLogic/fakedist-disk/alter_type (3.37s)
=== RUN   TestLogic/fakedist-disk
    --- FAIL: TestLogic/fakedist-disk (0.51s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #58269 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/type_change.go
+++ b/pkg/sql/type_change.go
@@ -209,6 +209,14 @@ func (t *typeSchemaChanger) exec(ctx context.Context) error {
 		}
 	}
 
+	// Make sure all of the leases have dropped before attempting to validate.
+	if err := WaitToUpdateLeases(ctx, leaseMgr, t.typeID); err != nil {
+		if errors.Is(err, catalog.ErrDescriptorNotFound) {
+			return nil
+		}
+		return err
+	}
+
 	// For all the read only members the current job is responsible for, either
 	// promote them to writeable or remove them from the descriptor entirely,
 	// as dictated by the direction.

